dockerfile/security/secret-in-build-arg.yaml

```diff
@@ -0,0 +1,24 @@
+rules:
+- id: secret-in-build-arg
+  patterns:
+  - pattern: ARG $ARG
+  - metavariable-regex:
+      metavariable: $ARG
+      regex: (?i).*(password|secret|token)
+  message: >-
+    Docker build time arguments are not suited for secrets, because the
+    argument values are saved with the image. Running `docker image history` on the
+    image will show information on how the image was built, including arguments. If
+    these contain secrets, anyone with access to the docker image can access those
+    secrets.
+  metadata:
+    category: security
+    technology:
+      - dockerfile
+    cwe: "CWE-538: Insertion of Sensitive Information into Externally-Accessible File or Directory"
```

Comment:
Would you mind adding a reference to the Owasp Top10 too?

```
      owasp:
        - A07:2021 - Identification and Authentication Failures
        - A03:2017 - Sensitive Data Exposure
```

Reply:
I couldn't really find an applicable entry in the top 10 of 2021. I don't think `A07:2021 - Identification and Authentication Failures` applies, really. But I'll add `A03:2017 - Sensitive Data Exposure`.

Reply:
The "What's changed..." section of https://owasp.org/Top10/ actually seems to suggest that `A03:2017 - Sensitive Data Exposure.` -> `A02:2021 Cryptographic Failures`: what do you think?

Reply:
Since there is no cryptography involved in either the problem or the solution, I would say _A02:2021 Cryptographic Failures_ is not a match.

The OWASP top 10 is not meant to be exhaustive. Some vulnerabilities fall into none of the categories.

Reply:
I added - A01:2021 - Broken Access Control